llvm/lib/Target/AMDGPU/SIFixSGPRCopies.cpp

```diff
@@ -1128,6 +1166,45 @@ void SIFixSGPRCopies::lowerVGPR2SGPRCopies(MachineFunction &MF) {
   }
 }
 
+void SIFixSGPRCopies::lowerPysicalSGPRInsts(MachineFunction &MF) {
+  for (auto &Entry : WaterFalls) {
+    MachineInstr *MI = Entry.first;
+    struct V2PysSCopyInfo Info = Entry.second;
+    if (Info.MOs.size() == 0 || Info.SGPRs.size() != Info.MOs.size())
```

Comment:
None of this can ever happen right? Is it worth checking them? Maybe put an assert?